### PR TITLE
native: thread fixes

### PR DESCRIPTION
**Pull request by latter-bolden**
- fixes missing nicknames in threads
- extracts upload image logic into reusable hook
- adds support for uploading images to threads
- updates parent post metadata when sending/receiving thread replies (reply count, reply authors, last reply time)

Fixes LAND-1852
Fixes LAND-1850

### CODE REVIEW

File: apps/tlon-mobile/src/hooks/useImageUpload.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import type {
```
Comment (patosullivan) [inline apps/tlon-mobile/src/hooks/useImageUpload.ts:1]: Nice

**Review by patosullivan — CHANGES_REQUESTED** [review]
Code looks good, but when I run locally I'm noticing that the reply counts can be off:


https://github.com/tloncorp/tlon-apps/assets/1221094/d4aa38d6-a9f7-4370-9b52-b260cc9474e8

**Comment by latter-bolden** [comment]
@patosullivan I think this was happening for sent ones because it was updating the parent reply meta both on send and upon hearing the fact. Should be fixed now

**Comment by patosullivan** [comment]
Might be off by one now?

https://github.com/tloncorp/tlon-apps/assets/1221094/13e639f7-0833-4ba3-bc2d-abf17e8d55df

**Review by patosullivan — APPROVED** [review]
I think this is good to go for now, but we'll have to watch to see if the (difficult to reproduce) off-by-one weirdness crops up again.

**Pull request merged**